### PR TITLE
Add LoadBalance

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

issues: #9481

### CODE REVIEW

File: internal/proto/query_coord.proto [hunk 1]
```diff
@@ -313,8 +314,11 @@ message HandoffSegmentsRequest {
 
 message LoadBalanceRequest {
   common.MsgBase base = 1;
-  repeated int64 source_nodeIDs = 2;
+  repeated int64 source_nodeIDs = 2; // deprecated
   TriggerCondition balance_reason = 3;
+  int64 src_nodeID = 6;
+  int64 dst_nodeID = 5;
+  repeated int64 sealed_segmentIDs = 4;
```
Comment (xige-16) [inline internal/proto/query_coord.proto:321]: Is it better to change repeated int64 sealed_segmentIDs to repeated SegmentInfo sealed_segments = 4?

File: internal/proto/query_coord.proto [hunk 2]
```diff
@@ -313,8 +314,11 @@ message HandoffSegmentsRequest {
 
 message LoadBalanceRequest {
   common.MsgBase base = 1;
-  repeated int64 source_nodeIDs = 2;
+  repeated int64 source_nodeIDs = 2; // deprecated
   TriggerCondition balance_reason = 3;
+  int64 src_nodeID = 6;
+  int64 dst_nodeID = 5;
```
Comment (xige-16) [inline internal/proto/query_coord.proto:320]: int64 src_nodeID = 6;
  int64 dst_nodeID = 5;
the number is wrong

File: internal/proto/query_coord.proto [hunk 3]
```diff
@@ -313,8 +314,11 @@ message HandoffSegmentsRequest {
 
 message LoadBalanceRequest {
```
Comment (xige-16) [inline internal/proto/query_coord.proto:315]: change to 
```
message LoadBalanceRequest {
  common.MsgBase base = 1;
  repeated int64 source_nodeIDs = 2;
  TriggerCondition balance_reason = 3;
  repeated int64 dst_nodeIDs = 4;
  repeated int64 sealed_segmentIDs = 5;
}
```

File: internal/proto/milvus.proto [hunk 4]
```diff
@@ -705,6 +706,16 @@ message GetMetricsResponse {
   string component_name = 3; // metrics from which component
 }
 
+/*
+* Do load balancing operation from src_nodeID to dst_nodeID.
+*/
+message LoadBalanceRequest {
```
Comment (xige-16) [inline internal/proto/milvus.proto:712]: change to 
```
message LoadBalanceRequest {
  common.MsgBase base = 1;
  int64 src_nodeID = 2;
  repeated int64 dst_nodeIDs= 3;
  repeated int64 sealed_segmentIDs = 4;
}
```

File: internal/proto/milvus.proto [hunk 5]
```diff
@@ -705,6 +706,16 @@ message GetMetricsResponse {
...
+  common.MsgBase base = 1;
+  int64 src_nodeID = 2;
+  repeated int64 dst_nodeID = 3;
```
Comment (xige-16) [inline internal/proto/milvus.proto:715]: repeated int64 dst_nodeIDs = 3;

File: internal/querycoord/impl.go [hunk 6]
```diff
@@ -544,6 +544,54 @@ func (qc *QueryCoord) GetSegmentInfo(ctx context.Context, req *querypb.GetSegmen
...
+		meta:               qc.meta,
+	}
+	err := qc.scheduler.Enqueue(loadBalanceTask)
+	if err != nil {
+		status.ErrorCode = commonpb.ErrorCode_UnexpectedError
+		status.Reason = err.Error()
+		return status, nil
+	}
+
+	err = loadBalanceTask.waitToFinish()
```
Comment (xige-16) [inline internal/querycoord/impl.go:581]: Synchronously waiting for loadbalance may time out

Comment (bigsheeper) [reply internal/querycoord/impl.go:581]: I think we should async it in task.go

**Pull request merged**